src/tree.rs

```diff
@@ -523,13 +523,26 @@ impl<'b> ParentBuilder<'b> {
     pub fn by_children(self, parent_guid: &Guid) -> Result<&'b mut Builder> {
         let parent_index = match self.0.entry_index_by_guid.get(parent_guid) {
             Some(&parent_index) if self.0.entries[parent_index].item.is_folder() => parent_index,
+            Some(&parent_index) => {
+                let parent = &self.0.entries[parent_index].item;
+
+                let child = match &self.1 {
+                    BuilderEntryChild::Exists(index) => &self.0.entries[*index].item,
+                    BuilderEntryChild::Missing(_guid) => unimplemented!(),
```

Comment:
Not entirely sure what I should be returning here for a missing child-- it looks like it should be a variant of `InvalidParent` where we don't know the child `Kind`.

Reply:
I think it would be fine to add new error variants. We certainly have a large number more in https://searchfox.org/mozilla-central/source/services/sync/modules/bookmark_validator.js#139-162. That said, many of those are somewhat redundant. @linacambridge, what do you think.

Reply:
A new variant sounds great to me! `InvalidParentForUnknownChild` and `MissingParentForUnknownChild`, maybe? That's a mouthful, but we don't match on the enum variants when we report errors, so I think it's all right.